src/column/reader.rs

```diff
@@ -615,8 +642,10 @@ mod tests {
     }
 
     fn add_values<T: DataType>(&mut self, encoding: Encoding, values: &[T::T]) where T: 'static {
-      assert!(self.num_values >= values.len() as u32, "num_values: {}, values.len(): {}",
-              self.num_values, values.len());
+      // when column is non-repeated, we should set all values in slice
+      if self.num_values < values.len() as u32 {
```

Comment:
Seems the `num_values` is set in `add_rep_levels`, which is problematic when `rep_level` is 0. Maybe we should pass the value for this from outside, perhaps using `levels_per_page`. What do you think?

Reply:
My reasoning was that we should not set num values in add_rep_levels function, but I also feel bad about removing the assertion in add_values method:). 

We should definitely keep assert. Could you elaborate a bit on your suggestion?

Reply:
`levels_per_page` ([see here](https://github.com/sadikovi/parquet-rs/blob/5e3f2f54beb6093d8fd5ce67ac63c6a2ac7fdecf/src/column/reader.rs#L677)) is the same as the number of values in the page (including nulls) I think. So we can pass that to the [ctor](https://github.com/sadikovi/parquet-rs/blob/5e3f2f54beb6093d8fd5ce67ac63c6a2ac7fdecf/src/column/reader.rs#L614) of `DatePageBuilderImpl` and assign that to `num_values`. Let me know if this makes sense. :)

Reply:
I think I understand. Will patch it, see what happens:)